code/game/objects/structures/tank_holder.dm

```diff
@@ -0,0 +1,134 @@
+/obj/structure/tank_holder
+	name = "tank holder"
+	desc = "A metallic frame that can hold tanks and extinguishers."
+	icon = 'icons/obj/tank.dmi'
+	icon_state = "holder"
+
+	custom_materials = list(/datum/material/iron = 2000)
+
+	density = FALSE
+	anchored = FALSE
+	pass_flags = LETPASSTHROW
+	max_integrity = 20
+
+	resistance_flags = FIRE_PROOF
+
+	var/obj/item/tank = null
+
+/obj/structure/tank_holder/Initialize()
+	. = ..()
+	if(tank)
+		density = TRUE
+		tank = new tank(src)
+
```

Comment:
needs atom_handle_del logic as well. (ie if an admin deletes the tank via vareditting, it'll set it back to the empty state)

```
/obj/structure/tank_holder/handle_atom_del(atom/A)
	if(A == tank)
		after_remove_tank()
	return ..()
```